presto-orc/src/main/java/com/facebook/presto/orc/stream/OrcInputStream.java

```diff
@@ -64,10 +57,11 @@ public OrcInputStream(OrcDataSourceId orcDataSourceId, FixedLengthSliceInput sli
 
         this.decompressor = requireNonNull(decompressor, "decompressor is null");
 
+        // memory reserved in the systemMemoryContext is never release and instead it is
+        // expected that the context itself will be destroyed at the end of the read
```

Comment:
`be destroyed at the end of the read` -> this statement is vague. 
Maybe we should say the aggregate or parent context will be closed when a stripe is completely read or sth like that.